### PR TITLE
feat: Added a function that draws the detector volume with inner surfaces

### PR DESCRIPTION
**Pull request by dimitra97**
Added the `drawDetectorVolumeWithSurfaces` function that draws the detector volume with the sub-volumes and their surfaces . Any comments and recommendations are more than welcome. 
@andiwand @Matthewharri @asalzburger

### CODE REVIEW

File: Core/include/Acts/Visualization/GeometryView3D.hpp [hunk 1]
```diff
@@ -114,6 +114,17 @@ struct GeometryView3D {
       const ViewConfig& connected = ViewConfig({0, 255, 0}),
       const ViewConfig& unconnected = ViewConfig({255, 0, 0}));
 
+  //Helper method to draw DetectorVolume with the inner surfaces 
+
+  static void drawDetectorVolumeWithSurfaces(
```
Comment (andiwand) [inline Core/include/Acts/Visualization/GeometryView3D.hpp:119]:
```suggestion
  /// Helper method to draw DetectorVolume with the inner surfaces 
  static void drawDetectorVolumeWithSurfaces(
```

File: Core/src/Visualization/GeometryView3D.cpp [hunk 2]
```diff
@@ -214,6 +214,32 @@ void Acts::GeometryView3D::drawDetectorVolume(
...
 }
 
+  void Acts::GeometryView3D::drawDetectorVolumeWithSurfaces(
+      IVisualization3D& helper,const Acts::Experimental::DetectorVolume& volume,
+      const GeometryContext& gctx,const Transform3& transform,
+      const ViewConfig& connected,const ViewConfig& unconnected, const ViewConfig& viewConfig){
+
+     
+     //draw the portals of the detectorvolume
+     auto portals = volume.portals();
```
Comment (andiwand) [inline Core/src/Visualization/GeometryView3D.cpp:224]:
```suggestion
     // draw the portals of the detector volume
     auto portals = volume.portals();
```

File: Core/src/Visualization/GeometryView3D.cpp [hunk 3]
```diff
@@ -214,6 +214,32 @@ void Acts::GeometryView3D::drawDetectorVolume(
...
+      drawSurface(helper, *surface, gctx, transform, viewConfig);
+     }
+     // draw the subvolumes with their surfaces using recursion
+     auto subvolumes = volume.volumes();
+     for(auto subvolume : subvolumes){
+        drawDetectorVolumeWithSurfaces(helper, *subvolume, gctx, transform, connected,
+                         unconnected, viewConfig);
+    }
+
+ }
```
Comment (andiwand) [inline Core/src/Visualization/GeometryView3D.cpp:241]:
```suggestion
 }
```

**Review by andiwand — COMMENTED** [review]
I wonder if we should combine this with `drawDetectorVolume` somehow or just create a new function as you did. maybe @Matthewharri has some input for us

other than that you should reformat the code using `clang-format` so it aligns with the rest of the code and the CI will stop complaining

**Comment by Matthewharri** [comment]
I think it might be best to have it combined with `drawDetectorVolume` somehow like @andiwand mentioned. But I do have a question, if you're drawing the portals, shouldn't that be all that is needed to draw the `DetectorVolume`? Or am I missing something obvious?

**Comment by dimitra97** [comment]
> I think it might be best to have it combined with `drawDetectorVolume` somehow like @andiwand mentioned. But I do have a question, if you're drawing the portals, shouldn't that be all that is needed to draw the `DetectorVolume`? Or am I missing something obvious?

**Comment by andiwand** [comment]
> I think it might be best to have it combined with `drawDetectorVolume` somehow like @andiwand mentioned. But I do have a question, if you're drawing the portals, shouldn't that be all that is needed to draw the `DetectorVolume`? Or am I missing something obvious?

thanks for the feedback @Matthewharri ! each detector volume has a list of portals and surfaces where the surfaces represent material or sensitive elements of the detector. is it okay for you if we plot all the surfaces all the time or should we flag it so you can also just plot the portals as it used to be

File: Core/include/Acts/Visualization/GeometryView3D.hpp [hunk 4]
```diff
@@ -103,16 +103,18 @@ struct GeometryView3D {
   /// @param [in,out] helper The visualization helper
   /// @param volume The DetectorVolume to be drawn
   /// @param gctx The geometry context for which it is drawn
+  /// @param drawSurface switches on/off the drawing of the volumes' surfaces
```
Comment (andiwand) [inline Core/include/Acts/Visualization/GeometryView3D.hpp:106]:
```suggestion
  /// @param drawSurfaces switches on/off the drawing of the volumes' surfaces
```

File: Core/include/Acts/Visualization/GeometryView3D.hpp [hunk 5]
```diff
@@ -103,16 +103,18 @@ struct GeometryView3D {
   /// @param [in,out] helper The visualization helper
   /// @param volume The DetectorVolume to be drawn
   /// @param gctx The geometry context for which it is drawn
+  /// @param drawSurfaces switches on/off the drawing of the volumes' surfaces
   /// @param transform An option additional transform
   /// @param connected The config for connected portals
   /// @param unconnected The config for unconnected portals
   static void drawDetectorVolume(
```
Comment (andiwand) [inline Core/include/Acts/Visualization/GeometryView3D.hpp:110]:
```suggestion
  /// @param unconnected The config for unconnected portals
  /// @param ViewConfig The drawing configuration
  static void drawDetectorVolume(
```

woops one more doc failure I overlooked

**Comment by andiwand** [comment]
@dimitra97 if you apply my suggested change we can merge this in

**Comment by dimitra97** [comment]
> @dimitra97 if you apply my suggested change we can merge this in

sure! i appplied

File: Core/include/Acts/Visualization/GeometryView3D.hpp [hunk 6]
```diff
@@ -103,16 +103,19 @@ struct GeometryView3D {
   /// @param [in,out] helper The visualization helper
   /// @param volume The DetectorVolume to be drawn
   /// @param gctx The geometry context for which it is drawn
+  /// @param drawSurfaces switches on/off the drawing of the volumes' surfaces
   /// @param transform An option additional transform
   /// @param connected The config for connected portals
   /// @param unconnected The config for unconnected portals
+  /// @param ViewConfig The drawing configuration
```
Comment (andiwand) [inline Core/include/Acts/Visualization/GeometryView3D.hpp:110]:
```suggestion
  /// @param viewConfig The drawing configuration
```

apparently this was upper case where I copied it from :facepalm: one more time @dimitra97

**Review by andiwand — APPROVED** [review]
:+1:

**Pull request merged**